kairos-server/src/config.rs

```diff
@@ -1,26 +1,119 @@
-use std::{fmt, str::FromStr};
+use config::{Config as Configuration, Environment, File};
+use dotenvy::dotenv;
+use serde::{Deserialize, Serialize};
+use std::{fs::File as FsFile, net::SocketAddr, path::Path};
+use tracing::{subscriber::set_global_default, Level};
+use tracing_subscriber::{fmt, prelude::*, EnvFilter};
 
-#[derive(Debug)]
-pub struct ServerConfig {
-    pub port: u16,
-}
+fn serialize_level_filter<S>(level: &Level, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    let level_str = match *level {
+        Level::ERROR => "error",
+        Level::WARN => "warn",
+        Level::INFO => "info",
+        Level::DEBUG => "debug",
+        Level::TRACE => "trace",
+    };
 
-impl ServerConfig {
-    pub fn from_env() -> Result<Self, String> {
-        let port = parse_env_as::<u16>("KAIROS_SERVER_PORT")?;
-        Ok(Self { port })
-    }
+    serializer.serialize_str(level_str)
 }
 
-fn parse_env_as<T>(env: &str) -> Result<T, String>
+fn deserialize_level_filter<'de, D>(deserializer: D) -> Result<Level, D::Error>
 where
-    T: FromStr,
-    <T as FromStr>::Err: fmt::Display,
+    D: serde::Deserializer<'de>,
 {
-    std::env::var(env)
-        .map_err(|e| format!("Failed to fetch environment variable {}: {}", env, e))
-        .and_then(|val| {
-            val.parse::<T>()
-                .map_err(|e| format!("Failed to parse {}: {}", env, e))
-        })
+    let s = String::deserialize(deserializer)?;
+    match s.to_lowercase().as_str() {
+        "error" => Ok(Level::ERROR),
+        "warn" => Ok(Level::WARN),
+        "info" => Ok(Level::INFO),
+        "debug" => Ok(Level::DEBUG),
+        "trace" => Ok(Level::TRACE),
+        _ => Err(serde::de::Error::custom("unknown level")),
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Server {
+    pub address: String,
```

Comment:
I think this should be a URL

Reply:
I guess yeah, do we have a crate with a Uri/Url type included? Are we planning on using the http crate?

Reply:
What is the type that axum expects?